CalendarApp/GoogleCalendarService/Manager/EventManager.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Google.Apis.Calendar.v3;
+using Google.Apis.Calendar.v3.Data;
+using Google.Apis.Requests;
+using GoogleCalendarService.Extensions;
+
+namespace GoogleCalendarService.Manager
+{
+    public class EventManager
+    {        
+        private readonly CalendarService _service;
+
+        public EventManager(CalendarService service)
+        {
+            _service = service;
+        }
+
+        //Need this for UpdateEventMethod
+        public List<Event> CheckEvent(string calendarId, Event eventForAdd, bool forcedEvent, string updateEventId)
+        {
+            eventForAdd.Id = updateEventId;
+            List<Event> conflictedEvents = CheckEvent(calendarId, eventForAdd, forcedEvent);
+            return conflictedEvents;
+        }
+
+        public List<Event> CheckEvent(string calendarId, Event eventForAdd, bool forcedEvent)
+        {
+            IList<Event> eventList = new List<Event>();
+            List<Event> conflictedEvents = new List<Event>();
+            if (forcedEvent)
+                return conflictedEvents;
+
+            EventPeriodicityHolder eventForAddHolder = new EventPeriodicityHolder(eventForAdd);
+
+            Events request = _service.Events.List(calendarId).Execute();
+            string token = request.NextPageToken;
+            while (token !=null)
+            {
+                var tempRequst = _service.Events.List(calendarId);
+                tempRequst.PageToken = token;
+                eventList = eventList.Concat(tempRequst.Execute().Items).ToList();
+                var updatedRequest = tempRequst.Execute();
+                token = updatedRequest.NextPageToken;
+            }
```

Comment:
I've seen similar logic in booking service. Can it be extracted and reused? With that logic extracted you can simplify the logic here and get rid of IList<Event> eventList = new List<Event>(); initialization